enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,546 @@
+storage in the host, with just the master Ignition as the handoff point.
+The BMC in the host can be used to automate
+deployment via a multi-cluster orchestration tool.
+
+The RHCOS live ISO image uses Ignition to configure the host, just as
+the RHCOS image used for running OpenShift does. This means Ignition
+is the most effective way to turn an RHCOS live image into a
+special-purpose image for performing the installation.
+
+We propose the following steps for deploying single-node instances of
+OpenShift:
+
+1. Add a new `create single-node-ignition-config` command to `openshift-installer`
+   which generates an Ignition config for a single-node deployment.
+2. Combine that Ignition config with an RHCOS live ISO image to build
+   an image for deploying OpenShift on a single node.
+3. Boot the new image on the host.
+4. Bootstrap the deployment, generating a new master Ignition config
+   and the static pod definitions for OpenShift. Write them, along
+   with an RHCOS image, to the disk in the host.
+5. Reboot the host to the internal disk, discarding the ephemeral live
+   image environment and allowing the previously generated artifacts
+   to complete the installation and bring up OpenShift.
+
+### User Stories
+
+#### As an OpenShift user, I want to be able to deploy OpenShift on a supported single node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints
+
+The installation images for single-node clusters will be unique for
+each cluster. The user or orchestration tool will create an
+installation image by combining the
+`bootstrap-in-place-for-live-iso.ign` created by the installer with an
+RHCOS live image using `coreos-install embed`. Making the image unique
+allows us to build on the existing RHCOS live image, instead of
+delivering a different base image, and means that the user does not
+need any infrastructure to serve Ignition configs to hosts during
+deployment.
+
+In order to add a viable, working etcd post reboot, we will take a
+snapshot of etcd and add it to the Ignition config for the host.
+After rebooting, we will use the restored `etcd-member` from the
+snapshot to rebuild the database. This allows etcd and the API service
+to come up on the host without having to re-apply all of the
+kubernetes operations run during bootstrapping.
+
+#### OpenShift-installer
+
+We will add a new `create single-node-ignition-config` command to the installer to create
+ `bootstrap-in-place-for-live-iso.ign` Ignition config.
+This new target will not output master.ign and worker.ign.
+Allow the user to specify the target disk drive for coreos-installer using environment
+ variable `OPENSHIFT_INSTALL_EXPERIMENTAL_BOOTSTRAP_IN_PLACE_COREOS_INSTALLER_ARGS`.
```

Comment:
There is some concern about settling on an API via install-config to enable the behavior until we're sure we understand all of the required inputs. Since we're focusing on zero-touch automated provisioning using the assisted installer workflow, there will only be one consumer of the feature for 4.8. After we have more runtime, we expect to have another enhancement (or update to this one) to describe a more public API via install-config.

Maybe @eranco74 and @staebler can provide more details?

Reply:
The EXPERIMENTAL was derived from comments that led to this enhancement: https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md
The assumption was that we want to follow same conventions for a preview feature.

Of course we can easily rename the ENV for now and address the install-config in 4.9

Reply:
Removed the env var and updated the alternatives.